### DISCUSSION TITLE
listenerMiddleware and RTK Query

### DISCUSSION THREAD

**Discussion by sbeigel** [reduxjs/redux-toolkit#4300] [Q&A] [answered]
I'm struggling to use createListenerMiddleware()'s startListening() to implement throttling for all PUT, POST and DELETE requests made with RTK Query. Even my simplest tests don't seem to work out:

matcher: api.endpoints.fooBar.matchPending, // or ...Fulfilled
effect: async (action, listenerApi) => {
    listenerApi.cancelActiveListeners()
    // or listenerApi.cancel() or listenerApi.delay(1000) ...
},

I'm thankful for any hints!

**Comment by EskiMojo14** [comment, marked as answer]
the listener middleware can't throttle anything from RTKQ. it's only called after actions are already dispatched, and the RTKQ mechanisms are already happening.

my suspicion is that throttling would need to be done via a custom base query.

**Reply by markerikson** [reply]
Additionally, the listener middleware can only _react_ to actions. It can't modify, pause, or delay them.
